src/OgContextInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Drupal\og;
+
+/**
+ * @file
+ * Contains \Drupal\og\OgContextInterface.
+ */
+
+use Drupal\Component\Plugin\PluginInspectionInterface;
+use Drupal\Core\Plugin\Context\ContextProviderInterface;
+
+/**
+ * Defines an interface for OG context plugins.
+ */
+interface OgContextInterface extends PluginInspectionInterface, ContextProviderInterface {
+
+  /**
+   * Return the current group. If no group will be found return Null.
+   *
+   * @return ContentEntityBase
+   *   Return the best matching group.
+   */
+  public function getGroup();
```

Comment:
The name of the class should be `OgContextProviderInterface` and it is the entry point to our plugins. We can have two methods:

* `getGroups` - will return all the groups found by the plugins.
* `getBestMatchingGroup` - will use a simple rule to get the "best matching" group